spec/formatting.md

```diff
@@ -23,11 +23,15 @@ are treated as identical.
 To resolve the value of a _variable_,
 its _name_ is used to identify either a local variable,
 or a variable defined elsewhere.
-If a local variable and an externally defined one use the same name,
-the local variable takes precedence.
 
-It is an error for a local variable definition to
-refer to a local variable that's defined after it in the message.
+It is an error for the right-hand side of a local variable declaration to
+refer to a local variable that's declared after it in the message.
+
+Variable names are required to be globally unique. That is,
```

Comment:
Use normative language.

```suggestion
Each `variable` MUST be unique within a message. It is a (what type?) error if a `declaration` uses a name assigned by a previous `declaration`.
```

Reply:
in the spirit of the generic comment I left on this PR, I suggest we use "constants" everywhere in this PR (and `const` keyword). (of course not for the passed-in ones)

Reply:
@mihnita Note that I'm using production names from the ABNF. If we decided to make the `variables` constant then we should change the name of the production at the same time.

Reply:
Rewrote this; for now I'll defer the `const` vs. `let` question and try to focus on making the language precise.